Adds MessageRecipient component

webinista opened the pull request:
- Closes department-of-veterans-affairs/messaging-fe#36

webinista committed:
Adds MessageRecipient component

- Closes department-of-veterans-affairs/messaging-fe#36

webinista committed:
Adds MessageRecipient component

- Closes department-of-veterans-affairs/messaging-fe#36

U-DON commented on src/js/messaging/reducers/compose.js:
```diff
@@ -1,17 +1,45 @@
+  SET_SUBJECT,
+  FETCH_RECIPIENTS_SUCCESS,
+  FETCH_RECIPIENTS_FAILURE
+} from '../actions/compose.js';
```
Can remove the .js suffix from imports.

U-DON commented on src/js/messaging/components/compose/MessageRecipient.jsx:
```diff
@@ -0,0 +1,41 @@
+      <div className="messaging-recipient">
+        <div className="messaging-recipient">
+          <ErrorableSelect
+              label="Recipient:"
```
Not `label="To: "`?

U-DON commented on src/js/messaging/actions/compose.js:
```diff
@@ -15,3 +20,21 @@ export function setSubject(value) {
   };
 }
+
+export function setRecipient(recipient) {
```
Maybe update the other `set*` function arguments to be consistent in style.

U-DON commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -4,13 +4,24 @@ import { connect } from 'react-redux';
+  setRecipient,
+  setSubject,
+  fetchRecipients
+} from '../actions/compose.js';
```
Don't need the .js suffix here (and for the config import above).

U-DON commented on src/js/messaging/components/compose/MessageRecipient.jsx:
```diff
@@ -0,0 +1,41 @@
+    const recipientValue = makeField(this.props.value);
+    return (
+      <div className="messaging-recipient">
+        <div className="messaging-recipient">
```
Same div but nested. Can this be removed?

U-DON commented on src/js/messaging/components/compose/MessageRecipient.jsx:
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ErrorableSelect from '../../../common/components/form-elements/ErrorableSelect';
+import { makeField } from '../../../common/model/fields.js';
```
Same about suffix in imports.

U-DON commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -22,6 +33,10 @@ class Compose extends React.Component {
     this.props.setSubject(valueObj);
   }
 
+  dispatchRecipientChange(event) {
```
Standardize on the argument name with the other `dispatch*` functions.

Although, on that topic, I'm not sure if `dispatch*` is the right name. Like for `dispatchCategoryChange`, it's actually calling two actions that have been bound to the dispatch. Perhaps `handle*`?

U-DON commented on src/js/messaging/containers/Compose.jsx:
```diff
@@ -30,6 +45,13 @@ class Compose extends React.Component {
+            value={this.props.compose.recipient}/>
         <MessageCategory
             categories={messageCategories}
             errorMessage={composeMessageErrors.category}
```
Below, correct typo to `composeMessageErrors.subject`.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#36

webinista committed:
Removing suffix from imports. Changing 'Recipient:' to 'To:'

webinista committed:
Removing suffix from imports. Changing 'Recipient:' to 'To:'

webinista committed:
Removing extraneous div element

webinista committed:
Removing link between category and subject line.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#36

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#36' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#36

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#36

webinista committed:
Consistently naming parameters in set* functions.

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#36' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#36

webinista commented:
This will have a lot of conflicts with #3059. Going to sit on this one for a little bit until that PR is merged.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#36

webinista committed:
Adds 'Confirm Refill' flow to Detail page. closes #3022

- Renamed SubmitButton to SubmitRefill.
- Made SubmitRefill into a form rather than a submit button.
- Removing redundant properties in SubmitRefill.
- Made openRefillModal part of props in Prescription.
- Moving loadPrescription, openGlossaryModal to props.
- Removed unnecessary name attribute from SubmitRefill.
- Adding rx-trigger class to make buttons that look like links.
- Changed glossary trigger to be a button instead of an `a` element for better compatibility with assistive tech.

alexose committed:
Education / training history (#3058)

* EducationHistory components.  still need to do conditional logic and tests

* add conditional logic and test

* add test

* isValidSection -> isValidPage

webinista committed:
Merge branch 'master' into issue#3022

jbalboni committed:
Move components into folders and refactor page container components (#3060)

* Move components into folders, remove page components

* Update tests

* Fix test

U-DON committed:
Merge branch 'master' into issue#3022

webinista committed:
Calling loadPrescription via this.props because undefined otherwise.

webinista committed:
Merge branch 'issue#3022' of github.com:department-of-veterans-affairs/vets-website into issue#3022

webinista committed:
De-links category, subject fields. (#3059)

* De-links category, subject fields.

WIP on department-of-veterans-affairs/messaging-fe#37

- Removed category radio buttons
- Updated CSS for new Subject line element group.
- Moved messaging-thread to partials directory
- Renamed message-category to _message-subject
- Renaming dispatch* methods to handle* methods.
- Added conditional require when the category field is set to 'Other'

* Adds ButtonDelete component. (#3048)

* Adds ButtonDelete component.

Closes department-of-veterans-affairs/messaging-fe#22

* Removing outline from delete button hover state.

* Adds ButtonDelete component.

Closes department-of-veterans-affairs/messaging-fe#22

* Removing outline from delete button hover state.

* Adding a new line to the end of messaging-btn-delete partial.

* Removing ButonDelete instances from Compose.jsx.

* Moved references to API URLs to project-specific configs. (#3049)

* Education / training history (#3058)

* EducationHistory components.  still need to do conditional logic and tests

* add conditional logic and test

* add test

* isValidSection -> isValidPage

* Move components into folders and refactor page container components (#3060)

* Move components into folders, remove page components

* Update tests

* Fix test

* Camel-casing fieldstate to fieldState.

* Moving app 'shell'-related CSS to a partial.

* De-links category, subject fields.

WIP on department-of-veterans-affairs/messaging-fe#37

- Removed category radio buttons
- Updated CSS for new Subject line element group.
- Moved messaging-thread to partials directory
- Renamed message-category to _message-subject
- Renaming dispatch* methods to handle* methods.
- Added conditional require when the category field is set to 'Other'

* Camel-casing fieldstate to fieldState.

* Moving app 'shell'-related CSS to a partial.

webinista committed:
Merge branch 'master' into issue#3022

webinista committed:
Merge pull request #3064 from department-of-veterans-affairs/issue#3022

Adds 'Confirm Refill' flow to Detail page. closes #3022

webinista committed:
Updating Compose.jsx with changes from master

- Renamed message-subject.scss to message-compose.scss

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#36' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#36

U-DON commented on src/js/messaging/components/compose/MessageRecipient.jsx:
```diff
@@ -0,0 +1,38 @@
+      React.PropTypes.shape({
+        label: React.PropTypes.string,
+        value: React.PropTypes.string }),
+    ])).isRequired
```
Props should be updated; most of these aren't being used, and you've added `value` and `options`.

U-DON commented on src/js/messaging/components/compose/MessageCategory.jsx:
```diff
@@ -9,26 +8,14 @@ class MessageCategory extends React.Component {
+            name="messageCategory"
+            onValueChange={this.props.onValueChange}
+            options={categories}
+            value={categoryValue}/>
```
Update `propTypes` below.

U-DON commented on src/js/messaging/components/compose/MessageSubject.jsx:
```diff
@@ -9,12 +9,13 @@ class MessageSubject extends React.Component {
+            additionalClass="messaging-subject-input"
             errorMessage={this.props.errorMessage}
-            label="Subject Line:"
+            label="Subject"
```
I still see "Subject Line: " in the latest (I think) mocks.

webinista replied:
"Subject Line" is in Compose.jsx as the `legend` for the `fieldset` that wraps this and the category field. This is the label for just the subject field.

U-DON commented on src/js/messaging/components/compose/MessageSubject.jsx:
```diff
@@ -26,6 +27,8 @@ MessageSubject.propTypes = {
   subjectClass: React.PropTypes.string,
   name: React.PropTypes.string,
```
Couple of unused props.

U-DON commented:
LGTM. Feel free to merge if I don't get to it after your changes.

webinista committed:
Making Compose sub-components use a cssClass property.

webinista committed:
Cleaning up props.

webinista commented:
Leaving the `errorMessage` prop in here because I'll have to add it later for validation. The error message needs to be set conditionally, triggered by `onblur` or `onsubmit`, otherwise the component will show an error state.

Tracking error checking in department-of-veterans-affairs/messaging-fe#55
